pkg/controller/volume/scheduling/scheduler_assume_cache.go

```diff
@@ -354,8 +355,9 @@ func pvStorageClassIndexFunc(obj interface{}) ([]string, error) {
 	return []string{""}, fmt.Errorf("object is not a v1.PersistentVolume: %v", obj)
 }
 
+// NewPVAssumeCache creates a PV assume cache.
 func NewPVAssumeCache(informer cache.SharedIndexInformer) PVAssumeCache {
-	return &pvAssumeCache{assumeCache: NewAssumeCache(informer, "v1.PersistentVolume", "storageclass", pvStorageClassIndexFunc)}
+	return &pvAssumeCache{NewAssumeCache(informer, "v1.PersistentVolume", "storageclass", pvStorageClassIndexFunc).(*assumeCache)}
```

Comment:
because it returns an interface now, we need to cast it to `*assumeCache`
note that we cannot use AssumeCache interface in `pvAssumeCache` struct because private functions of `assumeCache` are required in testing

Reply:
Hm can the unit tests do one more level of casting to get the 'assumeCache'?

Looking at L348, I'm not sure we need to copy private 'assumeCache' into the struct. It looks like we only use interface functions in the PV and PVC implementations

Reply:
> can the unit tests do one more level of casting to get the 'assumeCache'?

At before, I thought we need two variables to access pv/pvc)AssumeCache and assumeCache with which the code will be tedious, but I found in testing we didn't need to access private methods of (pv/pvc)AssumeCache.

It's a good idea. I pushed a new commit to address this: https://github.com/kubernetes/kubernetes/pull/77442/commits/4abd73048c35604a620a3cb6f68ffad6c42e3fe6